beluga_ros/test/test_point_cloud.cpp

```diff
@@ -0,0 +1,773 @@
+// Copyright 2024 Ekumen, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <memory>
+#include <vector>
+
+#include <range/v3/range/conversion.hpp>
+
+#include "beluga_ros/messages.hpp"
+#include "beluga_ros/point_cloud.hpp"
+
+namespace {
+
+auto make_message() {
+#if BELUGA_ROS_VERSION == 2
+  return std::make_shared<beluga_ros::msg::PointCloud2>();
+#elif BELUGA_ROS_VERSION == 1
+  return boost::make_shared<beluga_ros::msg::PointCloud2>();
+#endif
+}
+
+TEST(TestPointCloud, XYZPointsUnorderedPC) {
+  auto message = make_message();
+  const auto origin = Sophus::SE3d{};
+  message->width = 1;   // Unordered point cloud
+  message->height = 5;  // Number of points
+  // Set the point fields to x, y and z
+  int fields = 3;
+  message->fields.clear();
+  message->fields.reserve(fields);
+  // Set offset
+  int offset = 0;
+  offset = addPointField(*message, "x", 1, beluga_ros::msg::PointFieldF32, offset);
+  offset = addPointField(*message, "y", 1, beluga_ros::msg::PointFieldF32, offset);
+  offset = addPointField(*message, "z", 1, beluga_ros::msg::PointFieldF32, offset);
+  // Set message params
+  message->point_step = offset;
+  message->row_step = message->width * message->point_step;
+  message->is_dense = true;
+  message->data.resize(message->point_step * message->width * message->height);
+  // Create data iterators
+  beluga_ros::msg::PointCloud2Iterator<float> iter_x(*message, "x");
+  beluga_ros::msg::PointCloud2Iterator<float> iter_y(*message, "y");
+  beluga_ros::msg::PointCloud2Iterator<float> iter_z(*message, "z");
+  // Create some raw data for the points
+  const std::vector<float> point_data = {1.0f, 2.0f,  3.0f,  4.0f,  5.0f,  6.0f,  7.0f, 8.0f,
+                                         9.0f, 10.0f, 11.0f, 12.0f, 13.0f, 14.0f, 15.0f};
```

Comment:
@pvela2017 hmm, I think that using `std::vector<Eigen::Vector3f>` would make these tests easier to read. I have to think extra to identify `y` coordinates in your `point_data`.

Reply:
Changed d867fd3745f3540818b21c6dda282ca80d191689